src/ipc/ipc4/helper.c

```diff
@@ -990,7 +990,7 @@ const struct comp_driver *ipc4_get_comp_drv(int module_id)
 #if CONFIG_LIBRARY_MANAGER
 	if (!drv) {
 		/* New module not registered yet. */
-		lib_manager_register_module(desc, module_id);
+		lib_manager_register_module(module_id);
```

Comment:
see, here it's called `module_id`, and it's passed from two callers: from `ipc4_get_large_config_module_instance()` and from `comp_new_ipc4()` - in both cases as a module ID, in the latter case it's even as https://github.com/thesofproject/sof/blob/469c0a7e63d010eb5b7c13d1bc06084796beb7ed/src/ipc/ipc4/helper.c#L105 And it's different from "component ID" - it's missing the instance part. I still don't understand this renaming.

Reply:
In my opinion, `module_id` is ambiguous because it is not stated whether it contains only the module id or also the lib id.